fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+
+    private final Timer lakeTieringScheduleTimer;
+    private final Queue<Long> pendingTieringTables;
+    private final ExpiredOperationReaper expirationReaper;
+
+    // the live tables that are tiering
+    private final Set<Long> liveTieringTableIds;
+
+    // table_id -> table path
+    private final Map<Long, TablePath> tablePathById;
+
+    // table_id -> tiering interval
+    private final Map<Long, Long> tieringIntervalByTableId;
+
+    private final Lock lock = new ReentrantLock();
+
+    public LakeTableTieringManager() {
+        this(new DefaultTimer("delay lake tiering", 1_000, 20));
+    }
+
+    @VisibleForTesting
+    protected LakeTableTieringManager(Timer lakeTieringScheduleTimer) {
+        this.lakeTieringScheduleTimer = lakeTieringScheduleTimer;
+        this.pendingTieringTables = new ArrayDeque<>();
+        this.liveTieringTableIds = new HashSet<>();
+        this.tablePathById = new HashMap<>();
+        this.tieringIntervalByTableId = new HashMap<>();
+        this.expirationReaper = new ExpiredOperationReaper();
+        expirationReaper.start();
+    }
+
+    public void initWithLakeTables(List<TableInfo> lakeTables) {
+        for (TableInfo tableInfo : lakeTables) {
+            // try to schedule the tables to be tiered, we wait for a period of TIMEOUT
+            // to avoid the existing tiering services are still tiering it
+            addLakeTable(tableInfo, TIERING_SERVICE_TIMEOUT_MS);
+        }
+    }
+
+    public void addLakeTable(TableInfo tableInfo) {
+        addLakeTable(tableInfo, null);
+    }
+
+    private void addLakeTable(TableInfo tableInfo, @Nullable Long delayMs) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.put(tableInfo.getTableId(), tableInfo.getTablePath());
+                    tieringIntervalByTableId.put(
+                            tableInfo.getTableId(),
+                            tableInfo.getTableConfig().getDataLakeTieringInterval().toMillis());
+                    scheduleTableTiering(tableInfo.getTableId(), delayMs);
+                });
+    }
+
+    private void scheduleTableTiering(long tableId) {
+        scheduleTableTiering(tableId, null);
+    }
+
+    private void scheduleTableTiering(long tableId, @Nullable Long delayMs) {
+        Long tieringInterval = tieringIntervalByTableId.get(tableId);
+        if (tieringInterval == null) {
+            // the table has been dropped, return directly
+            return;
+        }
+        lakeTieringScheduleTimer.add(
+                new DelayedTiering(tableId, delayMs == null ? tieringInterval : delayMs));
+    }
+
+    public void removeLakeTable(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.remove(tableId);
+                    tieringIntervalByTableId.remove(tableId);
+                });
+    }
+
+    @Nullable
+    public Tuple2<Long, TablePath> requestTable() {
```

Comment:
This method is not thread-safe. `TimerTask` will modify the member fields in another thread. The implementation of `LakeTableTieringManager` should be thread-safe.

Reply:
Yes, it should be thread safe, forget to add lock.